libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/CohortStatsHeatmap.tsx

```diff
@@ -0,0 +1,147 @@
+  ICohortStatsHeatmapProps,
+  ICohortStatsHeatmapState
+> {
+  public static contextType = ModelAssessmentContext;
+  public context: React.ContextType<typeof ModelAssessmentContext> =
+    defaultModelAssessmentContext;
+
+  public render(): React.ReactNode {
+    const columns: string[] = [
+      localization.ModelAssessment.ModelOverview.countColumnHeader
+    ];
+    columns.push(
+      ...this.props.selectableMetrics
+        .filter((element) =>
+          this.props.selectedMetrics.includes(element.key.toString())
+        )
+        .map((element) => {
+          return element.text;
+        })
+    );
+
+    // generate table contents
+    const cohortLabeledStatistics = generateMetrics(
+      this.context.jointDataset,
+      this.props.cohorts.map((errorCohort) =>
+        errorCohort.cohort.unwrap(JointDataset.IndexLabel)
+      ),
+      this.context.modelMetadata.modelType
+    );
+
+    const items = generateCohortsStatsTable(
+      this.props.cohorts,
+      this.props.selectableMetrics,
+      cohortLabeledStatistics,
+      this.props.selectedMetrics
+    );
+
+    return (
+      <HeatmapHighChart
+        configOverride={{
+          chart: {
+            height: this.props.cohorts.length * 40 + 120,
+            type: "heatmap"
+          },
+          colorAxis: {
+            max: 1,
+            maxColor: "#1634F6",
+            min: 0,
+            minColor: "#FFFFFF"
+          },
+          legend: {
+            enabled: false
+          },
+          series: [
+            {
+              borderWidth: 1,
+              colorKey: "colorValue",
+              data: items,
+              dataLabels: {
+                enabled: true
+              },
+              name: "Metrics",
+              type: "heatmap"
+            }
+          ],
+          title: {
+            align: "left",
+            text: localization.ModelAssessment.ModelOverview
+              .dataCohortsHeatmapHeader
+          },
+          tooltip: {
+            formatter() {
+              // to avoid semantic error during build cast point to any
+              if (
+                this.point.y === undefined ||
+                (this.point as any).value === undefined ||
+                (this.point as any).value === null
+              ) {
+                return undefined;
+              }
+              const value = (this.point as any).value;
+              if (this.point.x === 0) {
+                // Count column
+                return localization.formatString(
+                  localization.ModelAssessment.ModelOverview.tableCountTooltip,
+                  this.series.yAxis.categories[this.point.y],
+                  value
+                );
+              }
+              // Metric columns
+              return localization.formatString(
+                localization.ModelAssessment.ModelOverview.tableMetricTooltip,
+                this.series.xAxis.categories[this.point.x].toLowerCase(),

```

Comment:
why do we only do toLowerCase() for xAxis, not yAxis?

Reply:
Great question! It's actually inverted, so x is y and y is x (just to make it a little more confusing). In any case, we put the metric name in lower case so that it renders with proper capitalization in a sentence. The y axis (or rather: x axis) on the other hand is the cohort name which may be capitalized on purpose. Those are decided by the user after all, so I felt like it's safer not to mess with the capitalization. I should really document this with a nice comment. Thanks for your question!